quickfixj-core/src/main/java/quickfix/FieldMap.java

```diff
@@ -669,7 +669,7 @@ public void replaceGroup(int num, Group group) {
     }
 
     public void removeGroup(int field) {
-        getGroups(field).clear();
+        this.groups.remove(field);
```

Comment:
Just a minor nitpick really: what about `getGroups().remove(field)`? Or at least get rid of the `this` because no other place in that class has it when accessing the `groups` field.